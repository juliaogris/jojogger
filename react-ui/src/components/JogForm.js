import React, { Component, PropTypes } from 'react'
import moment from 'moment'

export default class JogForm extends Component {
  render () {
    const { onCancel } = this.props
    return (
      <form>
        <label>Date</label>
        <input placeholder={moment().format('YYYY-MM-DD')} type='date' ref={c => (this.date = c)} />
        <label>Duration</label>
        <input placeholder={'01:38:12'} type='text' ref={c => (this.duration = c)} />
        <label>Distance in km</label>
        <input placeholder={'17.8'} type='text' ref={c => (this.distance = c)} />
        <button type='submit'>Save</button>
        <button onClick={onCancel}>Cancel</button>
      </form>
    )
  }
}

JogForm.PropTypes = {
  updateJog: PropTypes.bool.isRequired,
  onCancel: PropTypes.func.isRequired,
  setJogs: PropTypes.func.isRequired
}
