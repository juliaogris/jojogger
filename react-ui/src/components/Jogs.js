import React, { Component, PropTypes } from 'react'
import { DateRangePicker } from 'react-dates'
import Row from '../components/Row'
import Plus from '../svgs/Plus'
import '../css/datepicker.css'
import moment from 'moment'

export default class Jogs extends Component {
  constructor () {
    super()
    let focusedInput = null
    this.state = {
      focusedInput,
      jogs: [],
      startDate: null,
      endDate: null
    }
    this.onDatesChange = this.onDatesChange.bind(this)
  }

  onEdit (jogId) {
    console.log('onEdit', jogId)
  }

  onDatesChange ({ startDate, endDate }) {
    const between = j => moment(j.date).isBetween(startDate, endDate, 'day', '[]')
    const jogs = this.props.jogs.filter(between)
    this.setState({ startDate, endDate, jogs })
  }

  componentWillReceiveProps (nextProps) {
    const { jogs } = nextProps
    let startDate = null
    let endDate = null
    if (jogs.length > 0) {
      startDate = moment(jogs[0].date)
      endDate = moment(jogs[jogs.length - 1].date)
    }
    this.setState({ startDate, endDate, jogs })
  }

  render () {
    const { startDate, endDate, jogs } = this.state
    const renderRow = (jog) => (
      <Row
        jog={jog}
        onEdit={this.onEdit}
        id={jog.id}
        key={jog.id} />
    )
    return (
      <div className='jogs'>
        <DateRangePicker
          startDate={startDate}
          endDate={endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.focusedInput}
          onFocusChange={focusedInput => this.setState({ focusedInput })}
          numberOfMonths={1}
          customArrowIcon={<span className='drp'>→</span>}
          customCloseIcon={<span className='drp'>⨉</span>}
          showDefaultInputIcon
          showClearDates
          keepOpenOnDateSelect={false}
          isOutsideRange={day => !(day.isSameOrBefore(moment(), 'day'))}
          displayFormat='YYYY-MM-DD'
        />
        <table>
          <thead>
            <tr>
              <td>Date</td>
              <td>Distance</td>
              <td>Duration</td>
              <td>Speed</td>
              <td>
                <div className='add-button-wrap'>
                  <button className='add-button'><Plus className='add-icon' /></button>
                </div>
              </td>
            </tr>
          </thead>
          <tbody>
            { jogs.map(renderRow) }
          </tbody>
        </table>
      </div>
    )
  }
}

Jogs.propTypes = {
  jogs: PropTypes.array.isRequired
}
