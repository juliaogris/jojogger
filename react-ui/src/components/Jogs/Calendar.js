import React, { PropTypes, Component } from 'react'
import { DateRangePicker } from 'react-dates'
import '../../css/datepicker.css'
import moment from 'moment'

export default class Login extends Component {
  constructor () {
    super()
    this.state = { focusedInput: null }
  }
  render () {
    const { startDate, endDate, onDatesChange } = this.props
    return (
      <DateRangePicker
        startDate={startDate}
        endDate={endDate}
        onDatesChange={onDatesChange}
        focusedInput={this.state.focusedInput}
        onFocusChange={focusedInput => { this.setState({ focusedInput }) }}
        numberOfMonths={1}
        customArrowIcon={<span className='drp'>→</span>}
        customCloseIcon={<span className='drp'>⨉</span>}
        showDefaultInputIcon
        showClearDates
        keepOpenOnDateSelect={false}
        isOutsideRange={day => !(day.isSameOrBefore(moment(), 'day'))}
        displayFormat='YYYY-MM-DD'
          />
    )
  }
}
Login.propTypes = {
  onDatesChange: PropTypes.func.isRequired
}
