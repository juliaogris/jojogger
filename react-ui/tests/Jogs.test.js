/* global it expect describe */
import React from 'react'
import ReactDOM from 'react-dom'
import Jogs from '../src/components/Jogs'
import JogForm from '../src/components/Jogs/JogForm'
import renderer from 'react-test-renderer'

const sampleJogs = [
  { 'date': '2017-04-09', 'duration': '01:00:00', 'distance': 12.13, 'id': 'id100' },
  { 'date': '2017-04-08', 'duration': '01:00:00', 'distance': 11.5, 'id': 'id200' },
  { 'date': '2017-04-07', 'duration': '01:00:04', 'distance': 12, 'id': 'id1' },
  { 'date': '2017-04-06', 'duration': '01:00:00', 'distance': 10, 'id': 'id2' },
  { 'date': '2017-04-06', 'duration': '01:00:00', 'distance': 10, 'id': 'id3' },
  { 'date': '2017-04-06', 'duration': '01:00:00', 'distance': 10, 'id': 'id4' },
  { 'date': '2017-04-05', 'duration': '00:20:01', 'distance': 3.1, 'id': 'id5' },
  { 'date': '2017-04-05', 'duration': '01:32:00', 'distance': 6, 'id': 'id6' },
  { 'date': '2017-04-03', 'duration': '00:10:00', 'distance': 6, 'id': 'id7' },
  { 'date': '2017-04-02', 'duration': '01:00:00', 'distance': 10, 'id': 'id8' },
  { 'date': '2017-03-31', 'duration': '01:00:00', 'distance': 10, 'id': 'id9' }]

describe('Jogs', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Jogs
      onDatesChange={() => {}}
      setJogs={() => {}}
      jogs={[]}
      noJogs={false}
      user={{}}
    />, div)
  })

  it('renders "Jogs" and matches snapshot without filtered jogs', () => {
    const tree = renderer.create(<Jogs
      onDatesChange={() => {}}
      setJogs={() => {}}
      jogs={[]}
      noJogs={false}
      user={{}}
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders "Jogs" and matches snapshot with sample jogs', () => {
    const tree = renderer.create(<Jogs
      onDatesChange={() => {}}
      setJogs={() => {}}
      jogs={sampleJogs}
      noJogs={false}
      user={{}}
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders "Jogs" and matches snapshot without jogs', () => {
    const tree = renderer.create(<Jogs
      onDatesChange={() => {}}
      setJogs={() => {}}
      jogs={[]}
      noJogs
      user={{}}
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('JogForm', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<JogForm
      onCancel={() => {}}
    />, div)
  })

  it('renders "Jogs" and matches snapshot without jog', () => {
    const tree = renderer.create(<JogForm
      onCancel={() => {}}
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders "Jogs" and matches snapshot with jog', () => {
    const tree = renderer.create(<JogForm
      jog={{ duration: '01:34:00', date: '2016-12-18', distance: '12.3' }}
      onCancel={() => {}}
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
