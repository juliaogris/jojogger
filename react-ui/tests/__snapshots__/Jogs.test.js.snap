exports[`JogForm renders "Jogs" and matches snapshot with jog 1`] = `
<form
  action={[Function]}
  className="jojog-form">
  <label
    className="">
    Date
    <input
      className=""
      name="date"
      onChange={[Function]}
      type="date"
      value="2016-12-18" />
  </label>
  <label
    className="">
    Duration
    <input
      className=""
      name="duration"
      onChange={[Function]}
      type="text"
      value="01:34:00" />
  </label>
  <label
    className="">
    Distance in km
    <input
      className=""
      name="distance"
      onChange={[Function]}
      step="0.01"
      type="number"
      value="12.3" />
  </label>
  <div
    className="delete-row">
    <span
      className="delete-button"
      onClick={[Function]}>
      Delete this 
      Jog
    </span>
  </div>
  <div
    className="tick-button-wrap">
    <button
      className="tick-button"
      onClick={[Function]}
      type="submit">
      <svg
        className="add-icon"
        height="100%"
        viewBox="0 -1  41 42"
        width="100%"
        xmlns="http://www.w3.org/2000/svg">
        <g
          fill="none"
          fillRule="evenodd"
          stroke="none"
          strokeWidth="1">
          <polygon
            fill="white"
            fillRule="nonzero"
            points="14.5299498 13.3700502 15.9299498 11.9700502 26.9458691 22.4458691 20.9458691 28.4458691 19.5458691 27.0458691 24.1458691 22.4458691"
            transform="translate(20.737909, 20.207960) rotate(90.000000) translate(-20.737909, -20.207960) " />
        </g>
      </svg>
    </button>
  </div>
  <button
    className="cancel-button"
    onClick={[Function]}>
    ×
  </button>
</form>
`;

exports[`JogForm renders "Jogs" and matches snapshot without jog 1`] = `
<form
  action={[Function]}
  className="jojog-form">
  <label
    className="">
    Date
    <input
      className=""
      name="date"
      onChange={[Function]}
      type="date"
      value="2017-04-10" />
  </label>
  <label
    className="">
    Duration
    <input
      className=""
      name="duration"
      onChange={[Function]}
      type="text"
      value="01:00:00" />
  </label>
  <label
    className="">
    Distance in km
    <input
      className=""
      name="distance"
      onChange={[Function]}
      step="0.01"
      type="number"
      value={10} />
  </label>
  <div
    className="tick-button-wrap">
    <button
      className="tick-button"
      onClick={[Function]}
      type="submit">
      <svg
        className="add-icon"
        height="100%"
        viewBox="0 -1  41 42"
        width="100%"
        xmlns="http://www.w3.org/2000/svg">
        <g
          fill="none"
          fillRule="evenodd"
          stroke="none"
          strokeWidth="1">
          <polygon
            fill="white"
            fillRule="nonzero"
            points="14.5299498 13.3700502 15.9299498 11.9700502 26.9458691 22.4458691 20.9458691 28.4458691 19.5458691 27.0458691 24.1458691 22.4458691"
            transform="translate(20.737909, 20.207960) rotate(90.000000) translate(-20.737909, -20.207960) " />
        </g>
      </svg>
    </button>
  </div>
  <button
    className="cancel-button"
    onClick={[Function]}>
    ×
  </button>
</form>
`;

exports[`Jogs renders "Jogs" and matches snapshot with sample jogs 1`] = `
<div
  className="page">
  <div
    className="DateRangePicker">
    <div>
      <div
        className="DateRangePickerInput">
        <button
          aria-label="Focus on start date"
          className="DateRangePickerInput__calendar-icon"
          onClick={[Function]}
          type="button">
          <svg
            viewBox="0 0 1393.1 1500"
            xmlns="http://www.w3.org/2000/svg">
            <path
              d="M107.2 1392.9h241.1v-241.1H107.2v241.1zm294.7 0h267.9v-241.1H401.9v241.1zm-294.7-294.7h241.1V830.4H107.2v267.8zm294.7 0h267.9V830.4H401.9v267.8zM107.2 776.8h241.1V535.7H107.2v241.1zm616.2 616.1h267.9v-241.1H723.4v241.1zM401.9 776.8h267.9V535.7H401.9v241.1zm642.9 616.1H1286v-241.1h-241.1v241.1zm-321.4-294.7h267.9V830.4H723.4v267.8zM428.7 375V133.9c0-7.3-2.7-13.5-8-18.8-5.3-5.3-11.6-8-18.8-8h-53.6c-7.3 0-13.5 2.7-18.8 8-5.3 5.3-8 11.6-8 18.8V375c0 7.3 2.7 13.5 8 18.8 5.3 5.3 11.6 8 18.8 8h53.6c7.3 0 13.5-2.7 18.8-8 5.3-5.3 8-11.5 8-18.8zm616.1 723.2H1286V830.4h-241.1v267.8zM723.4 776.8h267.9V535.7H723.4v241.1zm321.4 0H1286V535.7h-241.1v241.1zm26.8-401.8V133.9c0-7.3-2.7-13.5-8-18.8-5.3-5.3-11.6-8-18.8-8h-53.6c-7.3 0-13.5 2.7-18.8 8-5.3 5.3-8 11.6-8 18.8V375c0 7.3 2.7 13.5 8 18.8 5.3 5.3 11.6 8 18.8 8h53.6c7.3 0 13.5-2.7 18.8-8 5.4-5.3 8-11.5 8-18.8zm321.5-53.6v1071.4c0 29-10.6 54.1-31.8 75.3-21.2 21.2-46.3 31.8-75.3 31.8H107.2c-29 0-54.1-10.6-75.3-31.8C10.6 1447 0 1421.9 0 1392.9V321.4c0-29 10.6-54.1 31.8-75.3s46.3-31.8 75.3-31.8h107.2v-80.4c0-36.8 13.1-68.4 39.3-94.6S311.4 0 348.3 0h53.6c36.8 0 68.4 13.1 94.6 39.3 26.2 26.2 39.3 57.8 39.3 94.6v80.4h321.5v-80.4c0-36.8 13.1-68.4 39.3-94.6C922.9 13.1 954.4 0 991.3 0h53.6c36.8 0 68.4 13.1 94.6 39.3s39.3 57.8 39.3 94.6v80.4H1286c29 0 54.1 10.6 75.3 31.8 21.2 21.2 31.8 46.3 31.8 75.3z" />
          </svg>
        </button>
        <div
          className="DateInput">
          <input
            aria-describedby=""
            aria-label="Start Date"
            autoComplete="off"
            className="DateInput__input needsclick"
            disabled={false}
            id="startDate"
            name="startDate"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Start Date"
            readOnly={false}
            required={false}
            type="text"
            value="" />
          
          <div
            className="DateInput__display-text">
            Start Date
          </div>
        </div>
        <div
          className="DateRangePickerInput__arrow">
          <span
            className="drp">
            →
          </span>
        </div>
        <div
          className="DateInput">
          <input
            aria-describedby=""
            aria-label="End Date"
            autoComplete="off"
            className="DateInput__input needsclick"
            disabled={false}
            id="endDate"
            name="endDate"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="End Date"
            readOnly={false}
            required={false}
            type="text"
            value="" />
          
          <div
            className="DateInput__display-text">
            End Date
          </div>
        </div>
        <button
          className="DateRangePickerInput__clear-dates DateRangePickerInput__clear-dates--hide"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          type="button">
          <span
            className="screen-reader-only">
            Clear Dates
          </span>
          <div
            className="DateRangePickerInput__close-icon">
            <span
              className="drp">
              ⨉
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
  <table>
    <thead>
      <tr>
        <td>
          Date
        </td>
        <td>
          Duration
        </td>
        <td>
          Distance
        </td>
        <td>
          Speed
        </td>
        <td>
          <div
            className="add-button-wrap">
            <button
              className="add-button"
              onClick={[Function]}>
              <svg
                className="add-icon"
                height="36px"
                viewBox="0 0 500 600"
                width="30px"
                xmlns="http://www.w3.org/2000/svg"
                xmlnsXlink="http://www.w3.org/1999/xlink">
                <path
                  d="M 241.99219,407.91016 L 241.99219,310.44922 L 144.72656,310.44922 L 144.72656,294.43359 L 241.99219,294.43359 L 241.99219,197.36328 L 257.61719,197.36328 L 257.61719,294.43359 L 355.27344,294.43359 L 355.27344,310.44922 L 257.61719,310.44922 L 257.61719,407.91016 L 241.99219,407.91016 z"
                  fill="white" />
              </svg>
            </button>
          </div>
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          2017-04-09
        </td>
        <td>
          1:00:00
        </td>
        <td>
          12.13
        </td>
        <td>
          12.13
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          2017-04-08
        </td>
        <td>
          1:00:00
        </td>
        <td>
          11.50
        </td>
        <td>
          11.50
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          2017-04-07
        </td>
        <td>
          1:00:04
        </td>
        <td>
          12.00
        </td>
        <td>
          11.99
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          2017-04-06
        </td>
        <td>
          1:00:00
        </td>
        <td>
          10.00
        </td>
        <td>
          10.00
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          2017-04-06
        </td>
        <td>
          1:00:00
        </td>
        <td>
          10.00
        </td>
        <td>
          10.00
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          2017-04-06
        </td>
        <td>
          1:00:00
        </td>
        <td>
          10.00
        </td>
        <td>
          10.00
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          2017-04-05
        </td>
        <td>
          20:01
        </td>
        <td>
          3.10
        </td>
        <td>
          9.29
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          2017-04-05
        </td>
        <td>
          1:32:00
        </td>
        <td>
          6.00
        </td>
        <td>
          3.91
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          2017-04-03
        </td>
        <td>
          10:00
        </td>
        <td>
          6.00
        </td>
        <td>
          36.00
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          2017-04-02
        </td>
        <td>
          1:00:00
        </td>
        <td>
          10.00
        </td>
        <td>
          10.00
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          2017-03-31
        </td>
        <td>
          1:00:00
        </td>
        <td>
          10.00
        </td>
        <td>
          10.00
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Jogs renders "Jogs" and matches snapshot without filtered jogs 1`] = `
<div
  className="page">
  <div
    className="DateRangePicker">
    <div>
      <div
        className="DateRangePickerInput">
        <button
          aria-label="Focus on start date"
          className="DateRangePickerInput__calendar-icon"
          onClick={[Function]}
          type="button">
          <svg
            viewBox="0 0 1393.1 1500"
            xmlns="http://www.w3.org/2000/svg">
            <path
              d="M107.2 1392.9h241.1v-241.1H107.2v241.1zm294.7 0h267.9v-241.1H401.9v241.1zm-294.7-294.7h241.1V830.4H107.2v267.8zm294.7 0h267.9V830.4H401.9v267.8zM107.2 776.8h241.1V535.7H107.2v241.1zm616.2 616.1h267.9v-241.1H723.4v241.1zM401.9 776.8h267.9V535.7H401.9v241.1zm642.9 616.1H1286v-241.1h-241.1v241.1zm-321.4-294.7h267.9V830.4H723.4v267.8zM428.7 375V133.9c0-7.3-2.7-13.5-8-18.8-5.3-5.3-11.6-8-18.8-8h-53.6c-7.3 0-13.5 2.7-18.8 8-5.3 5.3-8 11.6-8 18.8V375c0 7.3 2.7 13.5 8 18.8 5.3 5.3 11.6 8 18.8 8h53.6c7.3 0 13.5-2.7 18.8-8 5.3-5.3 8-11.5 8-18.8zm616.1 723.2H1286V830.4h-241.1v267.8zM723.4 776.8h267.9V535.7H723.4v241.1zm321.4 0H1286V535.7h-241.1v241.1zm26.8-401.8V133.9c0-7.3-2.7-13.5-8-18.8-5.3-5.3-11.6-8-18.8-8h-53.6c-7.3 0-13.5 2.7-18.8 8-5.3 5.3-8 11.6-8 18.8V375c0 7.3 2.7 13.5 8 18.8 5.3 5.3 11.6 8 18.8 8h53.6c7.3 0 13.5-2.7 18.8-8 5.4-5.3 8-11.5 8-18.8zm321.5-53.6v1071.4c0 29-10.6 54.1-31.8 75.3-21.2 21.2-46.3 31.8-75.3 31.8H107.2c-29 0-54.1-10.6-75.3-31.8C10.6 1447 0 1421.9 0 1392.9V321.4c0-29 10.6-54.1 31.8-75.3s46.3-31.8 75.3-31.8h107.2v-80.4c0-36.8 13.1-68.4 39.3-94.6S311.4 0 348.3 0h53.6c36.8 0 68.4 13.1 94.6 39.3 26.2 26.2 39.3 57.8 39.3 94.6v80.4h321.5v-80.4c0-36.8 13.1-68.4 39.3-94.6C922.9 13.1 954.4 0 991.3 0h53.6c36.8 0 68.4 13.1 94.6 39.3s39.3 57.8 39.3 94.6v80.4H1286c29 0 54.1 10.6 75.3 31.8 21.2 21.2 31.8 46.3 31.8 75.3z" />
          </svg>
        </button>
        <div
          className="DateInput">
          <input
            aria-describedby=""
            aria-label="Start Date"
            autoComplete="off"
            className="DateInput__input needsclick"
            disabled={false}
            id="startDate"
            name="startDate"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Start Date"
            readOnly={false}
            required={false}
            type="text"
            value="" />
          
          <div
            className="DateInput__display-text">
            Start Date
          </div>
        </div>
        <div
          className="DateRangePickerInput__arrow">
          <span
            className="drp">
            →
          </span>
        </div>
        <div
          className="DateInput">
          <input
            aria-describedby=""
            aria-label="End Date"
            autoComplete="off"
            className="DateInput__input needsclick"
            disabled={false}
            id="endDate"
            name="endDate"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="End Date"
            readOnly={false}
            required={false}
            type="text"
            value="" />
          
          <div
            className="DateInput__display-text">
            End Date
          </div>
        </div>
        <button
          className="DateRangePickerInput__clear-dates DateRangePickerInput__clear-dates--hide"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          type="button">
          <span
            className="screen-reader-only">
            Clear Dates
          </span>
          <div
            className="DateRangePickerInput__close-icon">
            <span
              className="drp">
              ⨉
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
  <table>
    <thead>
      <tr>
        <td>
          Date
        </td>
        <td>
          Duration
        </td>
        <td>
          Distance
        </td>
        <td>
          Speed
        </td>
        <td>
          <div
            className="add-button-wrap">
            <button
              className="add-button"
              onClick={[Function]}>
              <svg
                className="add-icon"
                height="36px"
                viewBox="0 0 500 600"
                width="30px"
                xmlns="http://www.w3.org/2000/svg"
                xmlnsXlink="http://www.w3.org/1999/xlink">
                <path
                  d="M 241.99219,407.91016 L 241.99219,310.44922 L 144.72656,310.44922 L 144.72656,294.43359 L 241.99219,294.43359 L 241.99219,197.36328 L 257.61719,197.36328 L 257.61719,294.43359 L 355.27344,294.43359 L 355.27344,310.44922 L 257.61719,310.44922 L 257.61719,407.91016 L 241.99219,407.91016 z"
                  fill="white" />
              </svg>
            </button>
          </div>
        </td>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`Jogs renders "Jogs" and matches snapshot without jogs 1`] = `
<p
  className="info">
  No jogs tracked.
  <button
    onClick={[Function]}>
    Add one.
  </button>
</p>
`;
